src/Nest/CommonAbstractions/SerializationBehavior/LazyDeserialize.cs

```diff
@@ -0,0 +1,14 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Nest
+{
+	public interface ILazyDeserialize
+	{
+		JsonSerializer Serializer { get; set; }
```

Comment:
Eh, yea I don't like the `ILazyDeserialize` at all.  I'm going to remove it and treat top hits as a special case in the converter instead.